### PR TITLE
feat: add leader election envionment variables to make them configurable

### LINKED ISSUE #3224: Expose leader election envionment variables to configure LeaseDuration、RenewDeadline and RetryPeriod
**Is your feature request related to a problem? Please describe.**
the leader election configurations like LeaseDuration、RenewDeadline and RetryPeriod can not be configured by envionment variables，we can not change the default value.

**Describe the solution you'd like**
We can expose those configurations via envionment variables, make them configurable.

### PR DESCRIPTION
**Pull request by ChenRussell**
Fix #3224 

add leader election envionment variables to make LeaseDuration、RenewDeadline and RetryPeriod configurable

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

**Comment by whynowy** [comment]
@ChenRussell - thanks, could you run `make codegen` to fix the ci?

**Comment by ChenRussell** [comment]
> @ChenRussell - thanks, could you run `make codegen` to fix the ci?

I can not fix this problem when I run `make codegen`:
```
./hack/generate-proto.sh
++++++++++++++++++++++++++++++++
++++ GENERATING PROTO FILES ++++
++++++++++++++++++++++++++++++++
package golang.org/x/tools/cmd/goimports: unrecognized import path "golang.org/x/tools/cmd/goimports": https fetch: Get "https://golang.org/x/tools/cmd/goimports?go-get=1": dial tcp 142.250.69.209:443: i/o timeout
```

File: controllers/cmd/start.go [hunk 1]
```diff
@@ -41,6 +42,20 @@ type ArgoEventsControllerOpts struct {
 	HealthPort       int32
 }
 
+func lookupEnvDurationOr(key string, o time.Duration) *time.Duration {
+	logger := logging.NewArgoEventsLogger().Named(eventbus.ControllerName)
+	v, found := os.LookupEnv(key)
+	if found && v != "" {
+		d, err := time.ParseDuration(v)
+		if err != nil {
+			logger.With(key, v).Fatalw("failed to parse", zap.Error(err))
```
Comment (ryancurrah) [inline controllers/cmd/start.go:51]:
```suggestion
			logger.With(key, v).Fatalw("parsing %s duration", key, zap.Error(err))
```

We already know it's a fatal error so no need to add failed, maybe more info like the suggestion would be more useful.

Comment (ChenRussell) [reply controllers/cmd/start.go:51]: Updated!

**Comment by ChenRussell** [comment]
@whynowy @ryancurrah @farcaller @EppO Could you review this PR? Thanks

**Pull request closed without merging**